tests/java_helper.rs

```diff
@@ -0,0 +1,399 @@
+            ("compression.type", compression),
+        ],
+    );
+
+    let producer = jvm
+        .create_instance(
+            "org.apache.kafka.clients.producer.KafkaProducer",
+            &[InvocationArg::from(props)],
+        )
+        .expect("creating KafkaProducer");
+
+    let mut futures = vec![];
+    for (topic_name, partition_index, record) in records {
+        let ts = (record.timestamp.unix_timestamp_nanos() / 1_000_000) as i64;
+        let k = String::from_utf8(record.key.unwrap()).unwrap();
+        let v = String::from_utf8(record.value.unwrap()).unwrap();
+
+        let headers = jvm
+            .create_instance(
+                "org.apache.kafka.common.header.internals.RecordHeaders",
+                &[],
+            )
+            .expect("creating KafkaProducer");
+        for (k, v) in record.headers {
+            jvm.invoke(
+                &headers,
+                "add",
+                &[
+                    InvocationArg::try_from(k).expect("key arg"),
+                    InvocationArg::from(to_java_bytes(&jvm, &v)),
+                ],
+            )
+            .expect("add header");
+        }
+
+        let jvm_record = jvm
+            .create_instance(
+                "org.apache.kafka.clients.producer.ProducerRecord",
+                &[
+                    InvocationArg::try_from(topic_name).expect("topic arg"),
+                    InvocationArg::try_from(partition_index).expect("partition arg"),
+                    InvocationArg::try_from(ts).expect("ts arg"),
+                    InvocationArg::try_from(k).expect("key arg"),
+                    InvocationArg::try_from(v).expect("value arg"),
+                    InvocationArg::from(headers),
+                ],
+            )
+            .expect("creating KafkaProducer");
+
+        let fut = jvm
+            .invoke(&producer, "send", &[InvocationArg::from(jvm_record)])
+            .expect("flush");
+        futures.push(fut);
+    }
+
+    jvm.invoke(&producer, "flush", &[]).expect("flush");
+
+    let mut offsets = vec![];
+    for fut in futures {
+        let record_metadata = jvm.invoke(&fut, "get", &[]).expect("polling future");
+        let record_metadata = jvm
+            .cast(
+                &record_metadata,
+                "org.apache.kafka.clients.producer.RecordMetadata",
+            )
+            .expect("cast to RecordMetadata");
+        let offset = jvm
+            .invoke(&record_metadata, "offset", &[])
+            .expect("getting offset");
+        let offset: i64 = jvm.to_rust(offset).expect("converting offset to Rust");
+        offsets.push(offset);
+    }
+
+    jvm.invoke(&producer, "close", &[]).expect("close");
+
+    offsets
+}
+
+/// Consume
+pub async fn consume(
+    connection: &str,
+    topic_name: &str,
+    partition_index: i32,
+    n: usize,
+) -> Vec<RecordAndOffset> {
+    let jvm = setup_jvm();
+
+    let props = create_properties(
+        &jvm,
+        &[
```

Comment:
Might want to set `enable.auto.commit` to `false`

Reply:
done